EtageControl/src/etage-control.ts

```diff
@@ -1,142 +1,165 @@
 import L from 'leaflet'
 import './etage-control.css'
 
+interface Building {
+  id: string
+  name: string
+  center: [number, number] // Latitude and longitude of the building
+  zoom: number
+}
+
+interface Level {
+  id: string
+  name: string
+  buildingId: string
+  center: [number, number] // We need this right now to use the same methods of the active option in the menu
+  zoom: number
+}
+
+// Using proper TS types for the options, including the custom ones.
+// To be verified.
+interface PlantquestEtageControlOptions extends L.ControlOptions {
+  buildings?: Building[]
+  levels?: Level[]
+}
+
 const PlantquestEtageControl = L.Control.extend({
   options: {
-    position: 'topright'
-  },
+    position: 'topright' // Default position, see this after in case of overlapping
```

Comment:
add a `debug:boolean` option as per https://github.com/plantquest/leaflet-plugins/pull/8 notes

Reply:
Done, added debug option and feature in the code.